tests/unit/Language/Pirouette/ExampleSpec.hs

```diff
@@ -17,30 +21,208 @@ canParseType _ = return ()
 canParseProgram :: PrtUnorderedDefs Ex -> Assertion
 canParseProgram _ = return ()
 
+canParseDefinition :: Definition Ex -> Assertion
+canParseDefinition _ = return ()
+
 tests :: [TestTree]
 tests =
   [ testCase "Can parse types" $ do
-      canParseType [ty| \x : Type . x |]
-      canParseType [ty| all f : Type -> Type . \(x : Type) (y : Type) . f x -> f y |]
+      canParseType [ty| \x : * . x |]
+      canParseType [ty| forall f : * -> * . \(x : *) (y : *) . f x -> f y |]
       canParseType [ty| One (Two Three) (Four Five) Six Seven |]
-      canParseType [ty| all a : Type . all b : Type . a -> b -> a |]
-      canParseType [ty| Integer -> (all a : Type . Integer) |]
-      canParseType [ty| all a : Type . a -> all b : Type . a -> Integer |]
-      canParseType [ty| all a : Type . a -> all b : (Type -> Type) . Integer -> b a -> Integer |],
+      canParseType [ty| forall a : * . forall b : * . a -> b -> a |]
+      canParseType [ty| Integer -> (forall a : * . Integer) |]
+      canParseType [ty| forall a : * . a -> forall b : * . a -> Integer |]
+      canParseType [ty| forall a : * . a -> forall b : (* -> *) . Integer -> b a -> Integer |],
     testCase "Type binders with or without parenthesis" $ do
-      let t1 = [ty| \(x : Type) (y : Type) (z : Type) . F x y |]
-          t2 = [ty| \(x : Type) . \(y : Type) . \(z : Type) . F x y |]
+      let t1 = [ty| \(x : *) (y : *) (z : *) . F x y |]
+          t2 = [ty| \(x : *) . \(y : *) . \(z : *) . F x y |]
        in (t1 :: Type Ex) @?= t2,
     testGroup
       "Can parse terms"
       [ testCase "Example 1" $
           canParseTerm [term| if @Integer 3 < 5 then 42 + 1 else 0 |],
         testCase "Example 2" $
-          canParseTerm [term| /\ (f : Type -> Type) (x : Type) . \k : f (Either x Bool) . fmap whatever k |]
+          canParseTerm [term| /\ (f : * -> *) (x : *) . \k : f (Either x Bool) . fmap whatever k |]
       ],
     testCase "Can parse program" $
       canParseProgram
         [prog|
-          data Either (a : Type) (b : Type) = Left : Either a b | Right : Either a b
+          data Either (a : *) (b : *) = Left : Either a b | Right : Either a b
+          main : Integer
+          main = 42
+        |],
+    testGroup
+      "New syntax for function declaration"
```

Comment:
While it is nice to have tests, we should watch out for too many tests. The test tree above is basically subsumed by this much more involved tests. Theonly one that might be worth keeping is the `"Type binders with or without parenthesis"` tests, which is a regression test from a previous bug on that.

Reply:
I agree the test tree should be pruned. However I think there are quite some other test cases that deserve to remain. They correspond to regression tests from issues during development of the parser. For example line folding.

Could we maybe lower the depth of the test tree and have only one combined test case for line folding, one for a complex case statement showcasing everything, one for function definition whose type has nested foralls with many variables and name shadowing?

Reply:
Test suite has been pruned a lot.